pymeasure/instruments/ametek/ametek7270.py

```diff
@@ -172,28 +208,49 @@ class Ametek7270(Instrument):
                                   """ Reads the input value of ADC4 in Volts """
                                   )
     id = Instrument.measurement("ID",
-                                """ Reads the instrument identification """
+                                """ Reads the instrument identification """,
+                                cast=str
                                 )
 
-    def __init__(self, adapter, name="Ametek DSP 7270", **kwargs):
+    def __init__(self, adapter, name="Ametek DSP 7270",
+                 read_termination='\x00',
+                 write_termination='\x00',
+                 **kwargs):
+
         super().__init__(
             adapter,
             name,
+            read_termination=read_termination,
+            write_termination=write_termination,
             **kwargs
         )
 
     def set_voltage_mode(self):
         """ Sets instrument to voltage control mode """
-        self.write("IMODE 0")
+        self.ask("IMODE 0")
+        self.sensitivity_values = self.SENSITIVITIES_IMODE[0]
 
     def set_differential_mode(self, lineFiltering=True):
         """ Sets instrument to differential mode -- assuming it is in voltage mode """
-        self.write("VMODE 3")
-        self.write("LF %d 0" % 3 if lineFiltering else 0)
+        self.ask("VMODE 3")
+        self.ask("LF %d 0" % 3 if lineFiltering else 0)
+
+    def set_current_mode(self, low_noise=False):
+        """ Sets instrument to current control mode with either low noise or high bandwidth"""
+        if low_noise:
+            self.ask("IMODE 2")
+            self.sensitivity_values = self.SENSITIVITIES_IMODE[2]
+        else:
+            self.ask("IMODE 1")
+            self.sensitivity_values = self.SENSITIVITIES_IMODE[1]
 
     def set_channel_A_mode(self):
         """ Sets instrument to channel A mode -- assuming it is in voltage mode """
-        self.write("VMODE 1")
+        self.ask("VMODE 1")
+
+    def identification(self):
+        """Get the instrument ID and firmware version"""
+        return f"{self.id}/{self.ask('VER')}"

```

Comment:
First I thought, you ment the "id" property, then I notice, that you use that property and add the firmware version.
Why does the ID + firmware version give another form of identification?

Reply:
well it is just a matter of better identification

Reply:
I'm still not convinced, that "identification" (the long form of ID) is a good name for that combination.

Do you need the firmware version (which could basically change) as an additional identification to the serial number?

Reply:
What do you think about `extended_identification` to state, that it is more than the mere identification, as the serial number is already the identification of the device.